src/processes/boundingBox.js

```diff
@@ -0,0 +1,44 @@
+module.exports = room => {
+  if (!room) {
+    const Room = require('@living-room/client-js')
+    room = new Room()
+  }
+
+  const box = [0.25, 0.25, 0.5, 0.5]
+
+  room.assert([
+    `draw a (255, 255, 255) line from (${box[0]}, ${box[1]}) to (${box[0] + box[2]}, ${box[1]})`,
+    `draw a (255, 255, 255) line from (${box[0] + box[2]}, ${box[1]}) to (${box[0] + box[2]}, ${box[1] + box[3]})`,
+    `draw a (255, 255, 255) line from (${box[0] + box[2]}, ${box[1] + box[3]}) to (${box[0]}, ${box[1] + box[3]})`,
+    `draw a (255, 255, 255) line from (${box[0]}, ${box[1] + box[3]}) to (${box[0]}, ${box[1]})`
+  ])
+
+  const bounce_off_wall = async function (changes) {
+    for (animal of changes.assertions) {
+      if (
+        animal.x <= box[0] ||
+        animal.y <= box[1] ||
+        animal.x >= box[2] ||
+        animal.y >= box[3]
+      ) {
+        room.retract(`${animal.name} has speed (${animal.dx}, ${animal.dy})`)
+        room.assert(`${animal.name} has speed (${-animal.dx}, ${-animal.dy})`)
```

Comment:
So this always reverses both directions, but I'd prefer if we fixed out if we need to reverse just one direction, e.g.
```js
const dx = (animal.x <= box[0] || animal.x >= box[2]) ? -animal.dx : animal.dx
const dy = (animal.y <= box[1] || animal.y >= box[3]) ? - animal.dy : animal.dy
if (dx != animal.dx || dy != animal.dy) {
  //...retractions/assertions here
}
```

Reply:
Good call